yaml-tests/src/test/java/MultiServerConnectionFactoryTest.java

```diff
@@ -42,61 +43,119 @@ void testDefaultPolicy(int initialConnection) throws SQLException {
         MultiServerConnectionFactory classUnderTest = new MultiServerConnectionFactory(
                 MultiServerConnectionFactory.ConnectionSelectionPolicy.DEFAULT,
                 initialConnection,
-                dummyConnectionFactory(),
-                List.of(dummyConnectionFactory()));
+                dummyConnectionFactory("Primary"),
+                List.of(dummyConnectionFactory("Alternate")));
         assertEquals(initialConnection, classUnderTest.getCurrentConnectionSelector());
-        MultiServerConnectionFactory.MultiServerRelationalConnection connection = (MultiServerConnectionFactory.MultiServerRelationalConnection)classUnderTest.getNewConnection(URI.create("Blah"));
-        assertEquals(0, connection.getCurrentConnectionSelector());
+
+        MultiServerConnectionFactory.MultiServerRelationalConnection connection =
+                (MultiServerConnectionFactory.MultiServerRelationalConnection)classUnderTest.getNewConnection(URI.create("Blah"));
         assertEquals(initialConnection, classUnderTest.getCurrentConnectionSelector());
+        assertConnection(connection, 0, "Primary");
+        RelationalPreparedStatement statement = connection.prepareStatement("SQL");
+        assertStatement(statement, "Primary");
+        statement = connection.prepareStatement("SQL");
+        assertStatement(statement, "Primary");
```

Comment:
It feels like it would be cleaner to inline the `statement` variable, so that this becomes more like:
```
        assertStatement(connection.prepareStatement("SQL"), "Primary");
        assertStatement(connection.prepareStatement("SQL"), "Primary");
```
and then you can more easily parse the alternations.
You could also streamline it a bit more to have:
```
assertStatement(connection, "Primary");
```

Reply:
Done (first option, wanted to keep statement visible at this level too)